knimin/lib/data_access.py

```diff
@@ -1402,6 +1402,60 @@ def getAGKitDetails(self, supplied_kit_id):
         else:
             return {}
 
+    def push_barcode_to_qiita_buffer(self, barcode):
+        """Adds barcode to the qiita buffer
+
+        Parameters
+        ----------
+        barcode : str
+            The identifier to stage
+        """
+        sql = """SELECT barcode
+                 FROM project_qiita_buffer"""
+        present = {i[0] for i in self._con.execute_fetchall(sql)}
+
+        if barcode in present:
+            return "Barcode in queue or already sent to Qiita"
+
+        else:
+            sql = """INSERT INTO project_qiita_buffer (barcode)
+                     VALUES (%s)"""
+            self._con.execute(sql, [barcode])
+            return "Barcode inserted"
+
+    def get_send_qiita_buffer_status(self):
+        """Obtain the present status of the Qiita submission buffer"""
+        sql = """SELECT state FROM project_qiita_buffer_status"""
+        return self._con.execute_fetchone(sql)
```

Comment:
This assumes that there is one and only one, is that OK?

Reply:
yes